soc/xtensa/intel_adsp/common/ipc.c

```diff
@@ -142,7 +150,7 @@ bool intel_adsp_ipc_send_message(const struct device *dev,
 	config->regs->idd = ext_data;
 	config->regs->idr = data | INTEL_ADSP_IPC_BUSY;
 	k_spin_unlock(&devdata->lock, key);
-	return true;
+	return 0;
```

Comment:
No, its not a mistake.
I've fixed all dependencies accepting value from this function.
I'm changing its signature in a4d954bdd01323bc62dfa5ec33910093306f5257 (previously was returning `bool`, `true` value on success. Now returns `int` and error code `0` on success).

Reply:
> I'm changing its signature in https://github.com/zephyrproject-rtos/zephyr/commit/a4d954bdd01323bc62dfa5ec33910093306f5257 (previously was returning bool, true value on success. Now returns int and error code 0 on success).

I've not been following the whole review sorry but this caught my eye. The return value of `intel_adsp_ipc_send_message()` is being used in SOF, more specifically in `src/drivers/intel/ace/ipc.c`, see SOF commit https://github.com/thesofproject/sof/commit/d5967d19132c#diff-19c1831368f785044dd1175a576d575cfa8c3e5496f66e67ee66e38d9b3a9ad3R152

So this is a backwards-incompatible change that will require careful synchronization in SOF.
BTW it would be good to submit this to `sof/west.yml` CI even before Zephyr merge if you haven't done yet, I mean not just because of this backwards-incompatible change.

PS: I think this comment should stay "unresolved" not because it's unresolved but to keep it visible because it's backwards-incompatible change.